source/common/network/io_socket_handle_impl.cc

```diff
@@ -0,0 +1,31 @@
+#include "common/network/io_socket_handle_impl.h"
+
+#include <errno.h>
+
+#include <iostream>
+
+#include "common/common/assert.h"
+
+using Envoy::Api::SysCallIntResult;
+using Envoy::Api::SysCallSizeResult;
+
+namespace Envoy {
+namespace Network {
+
+IoSocketHandleImpl::~IoSocketHandleImpl() {
+  if (fd_ != -1) {
+    close();
```

Comment:
I didn't think it was legal to call a virtual method in a destructor.  IoSocketHandle::close()?

Reply:
done